### ISSUE TITLE
independent form to send invites [feature request]

### ISSUE THREAD

**Issue by FurblandChannel** [snarfed/bridgy-fed#1421]
While the current invite system works, I think it’s a bit cumbersome. It would be nice to have it going alongside some sort of web-based form where you could input a username to send an invite to. (just a thought).

**Comment by snarfed** [comment]
Hmm! Interesting to hear you think that'd be easier than the current DM approach. Web form was actually our first idea, way back in Feb; background in https://github.com/snarfed/bridgy-fed/issues/880 (which was renamed) and https://github.com/snarfed/bridgy-fed/issues/880#issuecomment-2082865362 .

The catch is that I want to include the requesting user in the DM, and I want to authenticate them to prevent spoofing. DMs get me that for free, but for a web form, I'd need OAuth or something similar, which makes it a better project. Plus, OAuth is non-standard (ie off protocol) in both ATProto and ActivityPub, which makes it more awkward.

**Comment by FurblandChannel** [comment]
> Hmm! Interesting to hear you think that'd be easier than the current DM approach. Web form was actually our first idea, way back in Feb; background in #880 (which was renamed) and [#880 (comment)](https://github.com/snarfed/bridgy-fed/issues/880#issuecomment-2082865362) .
> 
> The catch is that I want to include the requesting user in the DM, and I want to authenticate them to prevent spoofing. DMs get me that for free, but for a web form, I'd need OAuth or something similar, which makes it a better project. Plus, OAuth is non-standard (ie off protocol) in both ATProto and ActivityPub, which makes it more awkward.

I know this is getting a bit outlandish, but perhaps accounts could be made and then linked to fedi accounts by means of the bot messaging you a code that you have to reproduce(?)

**Comment by snarfed** [comment]
Yeah that is pretty outlandish. 😆 DM is working pretty well so far. If I do this too, I expect it'd be a simple web form with OAuth.

**Comment by dos1** [comment]
> Interesting to hear you think that'd be easier than the current DM approach.

I can't send or receive DMs on the AP instance I'm at at all. Having a way to interact with the bot from an OAuthed web form would be very welcome.

**Comment by snarfed** [comment]
From https://github.com/snarfed/bridgy-fed/issues/1583 :

> I’ve heard people here mention how we need to verify accounts on the fediverse for various things time and time again. Bsky has a standard way of doing this, but most fedi platforms do not. I believe I have come up with a solution. It’s a bit janky, but it gets the job done.
> 
>     user visits page on fed.brid.gy and enters their handle
>     page supplies them with code and prompts user to check DMs
>     bot DMs user unique link which expires after a certain time period
>     user then has to enter the code from the other page into this one to verify their account
> 
> having to enter a code from the page that initiated the verification in the page at the unique link ensures an instance admin can’t just snoop around in people’s DMs and authenticate as them
